test/service/test_tdr.py

```diff
@@ -105,8 +106,11 @@ def test(dataset: BigQueryDataset):
             test(BigQueryDataset(project='test-project', name='name', is_snapshot=False))
 
     def test_tdr_dataset_config(self):
-        dataset = BigQueryDataset.parse(config.tdr_target)
-        self.assertNotEqual(dataset.is_snapshot, dataset.name.startswith('datarepo_'))
+        for target_type in ('snapshot', 'dataset'):
+            with mock.patch.dict(os.environ, AZUL_TDR_TARGET=f'tdr:project_name:{target_type}/target_name'):
+                with self.subTest(target_type=target_type):
+                    dataset = BigQueryDataset.parse(config.tdr_target)
```

Comment:
The elaborate mocking above is only to make config.tdr_target return a particular value. It seems like overkill.